src/modules/45-dashboards/pages/home/__tests__/CloneDashboardForm.test.tsx

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { act, fireEvent, render, RenderResult, screen, waitFor } from '@testing-library/react'
+import { TestWrapper } from '@common/utils/testUtils'
+import type { IDashboardFormData } from '@dashboards/types/DashboardTypes'
+import type { StringKeys } from 'framework/strings'
+import * as customDashboardServices from 'services/custom-dashboards'
+import CloneDashboardForm, { CloneDashboardFormProps } from '../CloneDashboardForm'
+
+const defaultProps: CloneDashboardFormProps = {
+  hideModal: jest.fn(),
+  reloadDashboards: jest.fn()
+}
+
+const renderComponent = (props: CloneDashboardFormProps): RenderResult => {
+  return render(
+    <TestWrapper>
+      <CloneDashboardForm {...props} />
+    </TestWrapper>
+  )
+}
+
+const mockEmptyGetFolderResponse: customDashboardServices.GetFolderResponse = {
+  resource: []
+}
+
+describe('CloneDashboardForm', () => {
+  beforeEach(() => {
+    jest
+      .spyOn(customDashboardServices, 'useGetFolder')
+      .mockImplementation(() => ({ data: mockEmptyGetFolderResponse, loading: false } as any))
+  })
+  afterEach(() => {
+    jest.spyOn(customDashboardServices, 'useCloneDashboard').mockReset()
+    jest.spyOn(customDashboardServices, 'useGetFolder').mockReset()
+  })
+
+  test('it should display Clone Dashboard Form', () => {
+    renderComponent(defaultProps)
+
+    const formTitle: StringKeys = 'dashboards.cloneDashboardModal.title'
+    expect(screen.getByText(formTitle)).toBeInTheDocument()
+  })
+
+  test('it should trigger callbacks and show toast upon successful submission', async () => {
+    const resourceIdentifier = '456'
+    const testFormData: IDashboardFormData = {
+      id: '123',
+      resourceIdentifier,
+      title: 'testTitle',
+      description: ''
+    }
+    const mockCallbackClone = jest.fn(() => Promise.resolve({ resource: { resourceIdentifier } }))
+    jest
+      .spyOn(customDashboardServices, 'useCloneDashboard')
+      .mockImplementation(() => ({ mutate: mockCallbackClone, loading: false } as any))
+    const mockCallbackHide = jest.fn()
+    const mockCallbackReload = jest.fn()
+
+    const testProps: CloneDashboardFormProps = {
+      hideModal: mockCallbackHide,
+      reloadDashboards: mockCallbackReload,
+      formData: testFormData
+    }
+
+    renderComponent(testProps)
+
+    const formButtonText: StringKeys = 'continue'
+    const formButton = screen.getByText(formButtonText)
+    expect(formButton).toBeInTheDocument()
+
+    act(() => {
+      fireEvent.click(formButton)
+    })
```

Comment:
Instead of using `fireEvent`, try `userEvent`. It's a higher-level thing that better mimics how a user actually behaves and comes already wrapped in an `act`, so you don't have to.
```suggestion
    userEvent.click(formButton)
```